pkg/middlewares/auth/forward.go

```diff
@@ -195,7 +217,41 @@ func (fa *forwardAuth) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	}
 
 	req.RequestURI = req.URL.RequestURI()
-	fa.next.ServeHTTP(rw, req)
+	fa.next.ServeHTTP(headers.NewResponseModifier(rw, req, cac.PostRequestModifyResponseHeaders), req)
+}
+
+func (c *copyAuthCookies) PostRequestModifyResponseHeaders(res *http.Response) error {
+	if len(c.authResponseCookieHeaders) > 0 {
+		setCookieHeaders := res.Header["Set-Cookie"]
+		res.Header.Del("Set-Cookie")
+
+		for _, setCookieString := range setCookieHeaders {
+			cookieName := getCookieName(setCookieString)
+			// only add, if not in forward auth response
+			if cookieName != "" && c.authResponseCookieHeaders[cookieName] == "" {
+				res.Header.Add("Set-Cookie", setCookieString)
+			}
+		}
+
+		for _, setCookieString := range c.authResponseCookieHeaders {
+			res.Header.Add("Set-Cookie", setCookieString)
+		}
+	}
+	return nil
```

Comment:
This is in addition to the code above. An alternative to this would be to gate this function with a check immediately at the start like this:

```go
        if len(c.authResponseCookieHeaders) == 0 {
		return nil
	}
```

```suggestion
	setCookieHeaders := res.Header["Set-Cookie"]
	res.Header.Del("Set-Cookie")

	for _, setCookieString := range setCookieHeaders {
		cookieName := getCookieName(setCookieString)
		// only add, if not in forward auth response
		if cookieName != "" && c.authResponseCookieHeaders[cookieName] == "" {
			res.Header.Add("Set-Cookie", setCookieString)
		}
	}

	for _, setCookieString := range c.authResponseCookieHeaders {
		res.Header.Add("Set-Cookie", setCookieString)
	}

	return nil
```